### PR TITLE
danna - Wrist

### PR DESCRIPTION
**Pull request by DANNA2828**
Not checked on robot yet.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.arcrobotics.ftclib.command.SubsystemBase;
+import com.arcrobotics.ftclib.hardware.ServoEx;
+import com.arcrobotics.ftclib.hardware.SimpleServo;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+public class Wrist extends SubsystemBase {
+
+    private final ServoEx servoWrist;
+    public WristState currentState;
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: the state need to be private

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.arcrobotics.ftclib.command.FunctionalCommand;
+import com.arcrobotics.ftclib.command.Subsystem;
+
+import org.firstinspires.ftc.teamcode.R;
+import org.firstinspires.ftc.teamcode.Robot;
+
+public class WristCommands {
+
+    public static FunctionalCommand moveToStateCommand(WristState state){
+
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java]: remove space

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+public class WristConstants {
+
+    public static String SERVO_HARDWARE_NAME = "wrist_servo";
+    public static double MIN_DEGREE = 0;
+    public static double MAX_DEGREE = 1;
+    public static double POSITION_TOLERANCE_DEGREES = 5;
+}
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java:9]: add space before last line

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        return new FunctionalCommand(
+                () -> Robot.getInstance().WRIST.setState(state),
+                () -> {},
+                interrupt -> Robot.getInstance().WRIST.stop(),
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java]: do you really want to stop at the end? why not keep aiming to target all the time?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+public class WristConstants {
+
+    public static String SERVO_HARDWARE_NAME = "wrist_servo";
+    public static double MIN_DEGREE = 0;
+    public static double MAX_DEGREE = 1;
+    public static double POSITION_TOLERANCE_DEGREES = 5;
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]: go again on the values of these constants please

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 6]
```diff
@@ -1,35 +1,29 @@
 package org.firstinspires.ftc.teamcode.subsystems.wrist;
 
 import com.arcrobotics.ftclib.command.SubsystemBase;
-import com.arcrobotics.ftclib.hardware.ServoEx;
-import com.arcrobotics.ftclib.hardware.SimpleServo;
 import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.Servo;
 
 public class Wrist extends SubsystemBase {
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java:7]: what does it get from SubsystemBase? why does it need to inherit it?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 7]
```diff
@@ -1,35 +1,29 @@
...
-                WristConstants.MIN_DEGREE,
-                WristConstants.MAX_DEGREE
-        );
+        this.wristServo = hardwareMap.servo.get(WristConstants.SERVO_HARDWARE_NAME);
     }
 
     public void setState(WristState state){
         currentState = state;
-        servoWrist.turnToAngle(currentState.positionDegrees);
+        wristServo.setPosition(currentState.positionZeroToOne);
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: I would've created a setPosition function in Wrist so we wouldn't need to access the servo wristServo and only the Wrist class

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 8]
```diff
@@ -3,8 +3,8 @@
 public class WristConstants {
 
     public static String SERVO_HARDWARE_NAME = "wrist_servo";
-    public static double MIN_DEGREE = 0;
-    public static double MAX_DEGREE = 1;
-    public static double POSITION_TOLERANCE_DEGREES = 5;
+    public static double INTAKE_POSITION = 0.2;
+    public static double SCORE_POSITION = 0.5;
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]: את אתמול:
erm..
i will just use the servo programmer to make it 1 and 0🤓☝

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 9]
```diff
@@ -1,35 +1,29 @@
...
     }
 
     public boolean isAtTargetState(WristState targetState){
-        return Math.abs(targetState.positionDegrees - servoWrist.getAngle()) < WristConstants.POSITION_TOLERANCE_DEGREES;
+        return Math.abs(targetState.positionZeroToOne - wristServo.getPosition()) < WristConstants.POSITION_TOLERANCE;
     }
 
     public void stop(){
-        servoWrist.turnToAngle(servoWrist.getAngle());
+        wristServo.setPosition(wristServo.getPosition());
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: cursed but i likey

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristState.java [hunk 10]
```diff
@@ -2,13 +2,12 @@
...
-    INTAKE(0),
-    CLIMB(50),
-    SCORE(100);
+    INTAKE(WristConstants.INTAKE_POSITION),
+    SCORE(WristConstants.SCORE_POSITION);
 
-    public final double positionDegrees;
-    WristState(double positionDegrees){
-        this.positionDegrees = positionDegrees;
+    public final double positionZeroToOne;
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristState.java]: לא אינדיקטיבי לגביי מה שזה עושה

**Review by ItayKauff — CHANGES_REQUESTED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristState.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+public enum WristState {
+
+    INTAKE(WristConstants.INTAKE_POSITION),
+    SCORE(WristConstants.SCORE_POSITION);
+
+    public final double targetPosition;
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristState.java:8]: i now understand what it means and i likey

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.Servo;
+
+public class Wrist extends SubsystemBase {
+
+    private final Servo wristServo;
+    private WristState currentState;
+
+    public Wrist(HardwareMap hardwareMap){
+        this.wristServo = hardwareMap.servo.get(WristConstants.SERVO_HARDWARE_NAME);
```
Comment (ItayKauff) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: add
currentState = <>;
and decide what will be the reset position

Comment (ItayKauff) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: so when the object constructs you will know what is the currentState

**Review by ItayKauff — CHANGES_REQUESTED** [review]

**Review by hillelv — COMMENTED** [review]
very nice - just delete unused imports and ctrl alt L

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/WristTest.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
+package org.firstinspires.ftc.teamcode.opmodes.testers;
+
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+
+import org.firstinspires.ftc.teamcode.Bindings;
+import org.firstinspires.ftc.teamcode.Robot;
+import org.firstinspires.ftc.teamcode.opmodes.DefaultRaz;
+
+@TeleOp(name = "Wrist Test")
+public class WristTest extends DefaultRaz {
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/WristTest.java]:
```suggestion
@TeleOp(name = "Raz Wrist Test")
public class RazWristTest extends DefaultRaz {
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/WristTest.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    @Override
+    public void execute() {
+        Robot.getInstance().WRIST.telemetry(telemetry);
+    }
+
+    @Override
+    public void configureBindings() {
+        Bindings.wristBindings(gamepad1, gamepad2);
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/WristTest.java]:
```suggestion
        Bindings.razWristTest(gamepad1, gamepad2);
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public boolean isAtTargetState(WristState targetState){
+        return Math.abs(targetState.targetPosition - getPosition()) < WristConstants.POSITION_TOLERANCE;
+    }
+
+    public void telemetry (Telemetry telemetry){
+        telemetry.addData("Wrist isAtTargetState?:", isAtTargetState(currentTargetState));
+        telemetry.addData("Wrist TargetState:", currentTargetState);
+    }
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: consider adding current position to telemtry

**Review by YHerm — COMMENTED** [review]
KEEP THE CONVENTION!!!

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Bindings.java [hunk 16]
```diff
@@ -1,10 +1,23 @@
...
+
+import org.firstinspires.ftc.teamcode.subsystems.wrist.WristCommands;
+import org.firstinspires.ftc.teamcode.subsystems.wrist.WristState;
 
 public class Bindings {
 
     private static GamepadEx MAIN_GAMEPAD;
     private static GamepadEx SECOND_GAMEPAD;
 
+    public static void wristBindings(Gamepad gamepad1, Gamepad gamepad2){
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Bindings.java]:
```suggestion
    public static void razWristTest(Gamepad gamepad1, Gamepad gamepad2){
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+import com.arcrobotics.ftclib.command.SubsystemBase;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.Servo;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+public class Wrist extends SubsystemBase {
+
+    private final Servo wristServo;
+    private WristState currentTargetState;
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: consider changing the name, current or target state?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+    private void setPosition(double position){
+        wristServo.setPosition(position);
+    }
+
+    private double getPosition(){
+        return wristServo.getPosition();
+    }
+
+    public void setState(WristState state){
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: can be protected

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+        currentTargetState = state;
+        setPosition(currentTargetState.targetPosition);
+    }
+
+    public boolean isAtTargetState(WristState targetState){
+        return Math.abs(targetState.targetPosition - getPosition()) < WristConstants.POSITION_TOLERANCE;
+    }
+
+    public void telemetry (Telemetry telemetry){
+        telemetry.addData("Wrist isAtTargetState?:", isAtTargetState(currentTargetState));
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]:
```suggestion
        telemetry.addData("Wrist isAtTargetState?: ", isAtTargetState(currentTargetState));
```

**Review by YHerm — COMMENTED** [review]
Merge master to you, ctrl alt l and remove unused imports.
REMEMBER: don't commit changes from git, fix all stuff from code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+        telemetry.addData("Wrist TargetState:", currentTargetState);
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]:
```suggestion
        telemetry.addData("Wrist TargetState: ", currentTargetState);
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.arcrobotics.ftclib.command.SubsystemBase;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.Servo;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+public class Wrist extends SubsystemBase {
+
+    private final Servo wristServo;
```
Comment (YHerm) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: you're in wrist class so consider renaming to just servo

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java [hunk 22]
```diff
@@ -18,9 +20,11 @@ public static Robot getInstance() {
     }
 
     //add subsystems...
+    public Wrist WRIST;
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java]: add get and make private

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+    private WristState currentTargetState;
+
+    public Wrist(HardwareMap hardwareMap){
+        this.wristServo = hardwareMap.servo.get(WristConstants.SERVO_HARDWARE_NAME);
+        currentTargetState = WristState.INTAKE;
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]: this.

**Review by greenblitz4590 — COMMENTED** [review]
merge master to you

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/RazWristTest.java [hunk 24]
```diff
@@ -0,0 +1,26 @@
+package org.firstinspires.ftc.teamcode.opmodes.testers;
+
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+
+import org.firstinspires.ftc.teamcode.Bindings;
+import org.firstinspires.ftc.teamcode.Robot;
+import org.firstinspires.ftc.teamcode.opmodes.DefaultRaz;
+
+@TeleOp(name = "Wrist Test")
+public class RazWristTest extends DefaultRaz {
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/RazWristTest.java]:
```suggestion
@TeleOp(name = "Raz Wrist Test")
public class RazWristTest extends DefaultRaz {
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 25]
```diff
@@ -0,0 +1,10 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+public class WristConstants {
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]: make protected

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 26]
```diff
@@ -0,0 +1,10 @@
...
+
+    public static String SERVO_HARDWARE_NAME = "wrist_gripper";
+    public static double INTAKE_POSITION = 1;
+    public static double SCORE_POSITION = 0;
+    public static double POSITION_TOLERANCE = 0.07;
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]: make protected

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java [hunk 27]
```diff
@@ -20,15 +20,20 @@ public static Robot getInstance() {
         return instance;
     }
 
-
+    private Wrist WRIST;
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java]: make small

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/RazWristTest.java [hunk 28]
```diff
@@ -0,0 +1,26 @@
+package org.firstinspires.ftc.teamcode.opmodes.testers;
+
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+
+import org.firstinspires.ftc.teamcode.Bindings;
+import org.firstinspires.ftc.teamcode.Robot;
+import org.firstinspires.ftc.teamcode.opmodes.DefaultRaz;
+
+@TeleOp(name = "Raz Wrist Test")
+public class RazWristTest extends DefaultRaz {
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/testers/RazWristTest.java:10]: line space

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java [hunk 29]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.arcrobotics.ftclib.command.FunctionalCommand;
+import com.arcrobotics.ftclib.command.Subsystem;
+
+import org.firstinspires.ftc.teamcode.R;
+import org.firstinspires.ftc.teamcode.Robot;
+
+public class WristCommands {
+
+    public static FunctionalCommand moveToStateCommand(WristState state){
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java]:
```suggestion
    public static FunctionalCommand moveToState(WristState state){
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java [hunk 30]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    protected void setState(WristState state){
+        targetState = state;
+        setPosition(targetState.targetPosition);
+    }
+
+    public boolean isAtTargetState(WristState targetState){
+        return Math.abs(targetState.targetPosition - getPosition()) < WristConstants.POSITION_TOLERANCE;
+    }
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/Wrist.java]:
```suggestion
    public boolean isAtTargetState(WristState targetState){
        return targetState.ticks == getPosition()
    }
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java [hunk 31]
```diff
@@ -0,0 +1,21 @@
...
+        return new FunctionalCommand(
+                () -> Robot.getInstance().getWrist().setState(state),
+                () -> {},
+                interrupt -> {},
+                () -> Robot.getInstance().getWrist().isAtTargetState(state),
+                Robot.getInstance().getWrist()
+        );
+    }
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java]: make instant command

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 32]
```diff
@@ -0,0 +1,10 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+class WristConstants {
+
+    static String SERVO_HARDWARE_NAME = "wrist_gripper";
+    static double INTAKE_POSITION = 1;
+    static double SCORE_POSITION = 0;
+    static double POSITION_TOLERANCE = 0.07;
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]: remove

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java [hunk 33]
```diff
@@ -0,0 +1,14 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+import com.arcrobotics.ftclib.command.FunctionalCommand;
+import com.arcrobotics.ftclib.command.InstantCommand;
+
+import org.firstinspires.ftc.teamcode.Robot;
+
+public class WristCommands {
+
+    public static InstantCommand moveToState(WristState state){
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java]: require

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 34]
```diff
@@ -0,0 +1,9 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+class WristConstants {
+
+    static String SERVO_HARDWARE_NAME = "wrist_gripper";
+    static double INTAKE_POSITION = 1;
+    static double SCORE_POSITION = 0;
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]: protected

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java [hunk 35]
```diff
@@ -0,0 +1,13 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+import com.arcrobotics.ftclib.command.InstantCommand;
+
+import org.firstinspires.ftc.teamcode.Robot;
+
+public class WristCommands {
+
+    public static InstantCommand moveToState(WristState state){
```
Comment (hillelv) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristCommands.java]:
```suggestion
    public static Command moveToState(WristState state){
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java [hunk 36]
```diff
@@ -0,0 +1,10 @@
+package org.firstinspires.ftc.teamcode.subsystems.wrist;
+
+class WristConstants {
+
+    protected static String SERVO_HARDWARE_NAME = "wrist_gripper";
```
Comment (hillelv) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/wrist/WristConstants.java]:
```suggestion
    protected static String SERVO_HARDWARE_ID = "wrist_gripper";
```

**Review by greenblitz4590 — APPROVED** [review]

**Review by ItayKauff — APPROVED** [review]

**Pull request merged**